code/model/SiteTree.php

```diff
@@ -2461,6 +2485,22 @@ public function doRevertToLive() {
 		}
 		
 		$this->invokeWithExtensions('onAfterRevertToLive', $this);
+		return true;
+	}
+
+	/**
+	 * Determine if this page references a parent which is archived, and not available in stage
+	 *
+	 * @return bool True if there is an archived parent
+	 */
+	protected function isParentArchived() {
+		if($parentID = $this->ParentID) {
+			$parentPage = Versioned::get_latest_version("SiteTree", $parentID);
+			if(!$parentPage || $parentPage->IsDeletedFromStage) {
```

Comment:
For whatever reason, the entirity of the code base uses the property accessor. I agree with your complaint, but this PR isn't really the place to change that standard. Let's make that a separate PR, and for now just keep this consistent.